server/src/main/java/org/elasticsearch/index/get/GetResult.java

```diff
@@ -222,20 +226,34 @@ public String sourceAsString() {
         return sourceAsMap();
     }
 
+
+    public Map<String, DocumentField> getMetadataFields() {
+        return metaFields;
+    }
+
+    public Map<String, DocumentField> getDocumentFields() {
+        return documentFields;
+    }
+
     public Map<String, DocumentField> getFields() {
+        Map<String, DocumentField> fields = new HashMap<>();
+        fields.putAll(metaFields);
+        fields.putAll(documentFields);
         return fields;
     }
 
     public DocumentField field(String name) {
-        return fields.get(name);
+        return getFields().get(name);
     }
 
     @Override
     public Iterator<DocumentField> iterator() {
-        if (fields == null) {
+        // need to join the fields and metadata fields
+        Map<String, DocumentField> allFields = this.getFields();
+        if (allFields == null) {
```

Comment:
I dont' think this is possible? `getFields()` never returns null

Reply:
done